src/validation.cpp

```diff
@@ -1787,8 +1787,8 @@ DisconnectResult CChainState::DisconnectBlock(const CBlock& block, const CBlockI
                 error("DisconnectBlock(): transaction and undo data inconsistent");
                 return DISCONNECT_FAILED;
             }
-            for (unsigned int j = tx.vin.size(); j-- > 0;) {
-                const COutPoint &out = tx.vin[j].prevout;
+            for (int j = int(tx.vin.size()); j-- > 0;) {
```

Comment:
Is this equivalent to below? If so, is that more clear?
```suggestion
            for (int j = int(tx.vin.size()) - 1; j >= 0; j--) {
```

Reply:
pushed something else